Cython/Compiler/Optimize.py

```diff
@@ -2312,6 +2312,35 @@ def _optimise_generic_builtin_method_call(self, node, attr_name, function, arg_l
         return ExprNodes.CachedBuiltinMethodCallNode(
             node, function.obj, attr_name, arg_list)
 
+    PyObject_String_func_type = PyrexTypes.CFuncType(
+        Builtin.str_type, [
+            PyrexTypes.CFuncTypeArg("org", PyrexTypes.py_object_type, None)
+            ])
+
+    def _handle_simple_function_str(self, node, function, pos_args):
+        """Optimize single argument calls to str().
+        """
+        if len(pos_args) != 1:
+            if len(pos_args) == 0:
+                return ExprNodes.StringNode(node.pos, value=EncodedString(), constant_result='')
+            return node
+        arg = pos_args[0]
+
+        if basestring is str:  # python 2
```

Comment:
You're definitely interested in what the generated file is compiled in, rather than what version of Python Cython is running in. You should be able to Cythonize a file in Py2 then compile it for Py3

Reply:
Makes sense. Is there a canonical way to detect this? Looking at the unicode impl I think it was doing it by checking if unicode is a builtin, which maybe could do here as well (just felt out of place)

Reply:
The `PY_MAJOR_VERSION` macro in the C file. But I'm not convinced you actually need it - the [linked issue implies](https://github.com/cython/cython/issues/3279) that `PyObject_Str` should generally be OK.